jyulb/internal/isothermal/jyu_engine.py

```diff
@@ -313,51 +235,31 @@ def iter_lattices(self, names=None):
             names of specific lattices to be iterated over. If names is not
             given, then all lattices will be iterated over.
 
-        Yields
-        -------
-        ABCLattice
-        """
-        yield self._lat
-
-    def iter_meshes(self, names=None):
-        """Iterate over a subset or all of the meshes.
-
-        Parameters
-        ----------
-        names : sequence of str, optional
-            names of specific meshes to be iterated over. If names is not
-            given, then all meshes will be iterated over.
-
-        Yields
+        Returns
         -------
-        ABCMesh
+        A generator of ABCLattice objects
 
         Raises
         ------
-        NotImplementedError
-           always.
+        ValueError
+            if any one of the names is not in SD
         """
-        message = 'JYUEngine does not handle meshes'
-        raise NotImplementedError(message)
-
-    def iter_particles(self, names=None):
-        """Iterate over a subset or all of the particle containers.
+        self._update_dataset_names()
+        if names is None:
+            for name in self.SD.keys():
+                yield self.SD[name]
+        else:
+            for name in names:
+                if name not in self.SD.keys():
+                    message = 'State data does not contain requested item'
+                    raise ValueError(message)
+                yield self.SD[name]
+
+    def _update_dataset_names(self):
```

Comment:
> If you just have a single lattice, why not just use a single instance variable.

A wrapper should have SD attribute defined by the specs, and actually using SD to store proxy objects then makes sense to me. Also, it could be possible in future to modify JyuLB to support multiple datasets, so it's good to have SD structure ready for it. That was the reason I dropped _lattice (or similar: I don't remember now what it was called) which was used for storing the lattice in earlier versions.

I would prefer to use _SD dictionary in a way it will be used with other wrappers. Moving to multiple datasets support will be easier in the case of JyuLB. Also, writing new common templates for engines will be easier when "finding a common denominator" is more straightforward.